core/trino-spi/src/main/java/io/trino/spi/type/TypeOperators.java

```diff
@@ -181,10 +189,14 @@ public OperatorAdaptor(ScalarFunctionAdapter functionAdapter, OperatorConvention
             this.operatorConvention = operatorConvention;
         }
 
-        public synchronized MethodHandle get()
+        // optimize the read lock caused by synchronized
+        // public synchronized MethodHandle get()
+        public MethodHandle get()
         {
             if (adapted == null) {
```

Comment:
`adapted` would have to be declared `volatile`, no?

Reply:
Hello, this is to avoid locking when reading data. During the TB level data test, it is found that this code has a blocked lock. If modified, it can provide certain performance when reading data

Reply:
Yep. But it still needs to be `volatile` to be correct. Since that has a slight performance implication (a memory barrier for each access), it might be prudent to redo the perf test.